### PR TITLE
Integrated progenerated bulletproof , And Mixer deposit

### PR DESCRIPTION
**Pull request by AhmedKorim**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -3670,81 +3670,69 @@
     "@webassemblyjs/wast-parser" "1.8.5"
     "@xtuc/long" "4.2.2"
 
-"@webb-tools/api@0.1.2-18", "@webb-tools/api@^0.1.2-22", "@webb-tools/api@^0.1.2-34", "@webb-tools/api@^0.1.2-37":
-  version "0.1.2-37"
-  resolved "https://registry.yarnpkg.com/@webb-tools/api/-/api-0.1.2-37.tgz#aff4113c6e80d79932f7dd5d2a3bcc729665abe6"
-  integrity sha512-1mu8eKnz7nq8GqRJrM28tfJKGtjtJJ2Ga9fEeNazPSuBgharwRczc1rUPComS6cFl2mbzlECLt9UkXijpUWJCA==
+"@webb-tools/api@0.1.2-18", "@webb-tools/api@^0.1.2-22", "@webb-tools/api@^0.1.2-41", "@webb-tools/api@file:../webb.js/packages/api/build":
+  version "0.1.2-41"
```
Comment (drewstone) [inline yarn.lock:3674]: @AhmedKorim maybe you need to run `yarn` again? All versions points to 41 even though from above it looks like we should be on 42. Also, please remove resolutions when you are submitting updates to the lock file.

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good, some minor changes before merging.

File: packages/react-environment/src/MixerProvider.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ export interface MixerContextData {
   mixer: Mixer | null;
   shouldDestroy: false;
   init(): Promise<void>;
+  generatingBP: boolean;
```
Comment (drewstone) [inline packages/react-environment/src/MixerProvider.tsx:11]: can we use full description here? `generatingBulletproof`

File: packages/react-hooks/src/mixer/useMixer.ts [hunk 3]
```diff
@@ -46,14 +47,46 @@ export const useMixer = () => {
   const mixerInfos = useMixerInfos();
 
   const [mixerResult, setMixerResult] = useState<Omit<MixerContextData, 'init'>>({
+    generatingBP: false,
```
Comment (drewstone) [inline packages/react-hooks/src/mixer/useMixer.ts:50]: `generatingBulletproof`

File: packages/react-hooks/src/mixer/useMixer.ts [hunk 4]
```diff
@@ -46,14 +47,46 @@ export const useMixer = () => {
...
     initialized: false,
     loading: false,
     mixer: null,
     shouldDestroy: false,
   });
-
+  const [bulletproofGens, setBulletproofGens] = useLocalStorage('bulletproof_gens');
+  const generateBP = useCallback(async () => {
```
Comment (drewstone) [inline packages/react-hooks/src/mixer/useMixer.ts:57]: `generateBulletproof`

Comment (drewstone) [reply packages/react-hooks/src/mixer/useMixer.ts:57]:
```suggestion
  const generateBulletproof = useCallback(async () => {
```

File: packages/react-hooks/src/mixer/useMixer.ts (hunk 3 above)
Comment (drewstone) [reply packages/react-hooks/src/mixer/useMixer.ts:50]:
```suggestion
    generatingBulletproof: false,
```

File: packages/react-environment/src/MixerProvider.tsx (hunk 2 above)
Comment (drewstone) [reply packages/react-environment/src/MixerProvider.tsx:11]:
```suggestion
  generatingBulletproof: boolean;
```

File: packages/react-hooks/src/mixer/useMixer.ts [hunk 5]
```diff
@@ -46,14 +47,46 @@ export const useMixer = () => {
...
+      const gens = encoder.encode(bulletproofGens);
+      logger.info(`Encoded Bulletproof from localstorage string`);
+      return gens;
+    } else {
+      logger.info(`Generating Bulletproof`);
+    }
+    const worker = new Worker();
+    setMixerResult((p) => ({
+      ...p,
+      generatingBP: true,
```
Comment (drewstone) [inline packages/react-hooks/src/mixer/useMixer.ts:71]:
```suggestion
      generatingBulletproof: true,
```

File: packages/react-hooks/src/mixer/useMixer.ts [hunk 6]
```diff
@@ -46,14 +47,46 @@ export const useMixer = () => {
...
+    }));
+    logger.trace(`Generating Bulletproof`);
+    const pBG = await Mixer.preGenerateBulletproofGens(worker).finally(() => {
+      setMixerResult((p) => ({
+        ...p,
+        generatingBP: true,
```
Comment (drewstone) [inline packages/react-hooks/src/mixer/useMixer.ts:77]:
```suggestion
        generatingBulletproof: true,
```

File: packages/react-hooks/src/mixer/useMixer.ts [hunk 7]
```diff
@@ -46,14 +47,46 @@ export const useMixer = () => {
...
+    } else {
+      logger.info(`Generating Bulletproof`);
+    }
+    const worker = new Worker();
+    setMixerResult((p) => ({
+      ...p,
+      generatingBP: true,
+    }));
+    logger.trace(`Generating Bulletproof`);
+    const pBG = await Mixer.preGenerateBulletproofGens(worker).finally(() => {
```
Comment (drewstone) [inline packages/react-hooks/src/mixer/useMixer.ts:74]: Please make this more verbose.

File: packages/react-hooks/src/mixer/useMixer.ts [hunk 8]
```diff
@@ -70,8 +103,9 @@ export const useMixer = () => {
     }));
 
     try {
-      const mixer = await Mixer.init(new Worker(), mixerIds);
-
+      let gens = await generateBP();
```
Comment (drewstone) [inline packages/react-hooks/src/mixer/useMixer.ts:106]:
```suggestion
      let gens = await generateBulletproof();
```

**Review by drewstone — CHANGES_REQUESTED** [review]
Would also appreciate more comments here @AhmedKorim and always err on being verbose in variable names so that its clear for other readers.

File: yarn.lock (hunk 1 above)
Comment (AhmedKorim) [reply yarn.lock:3674]: this is from sub-packages, the version here is obsoleted by yarn resolution

**Pull request merged**